src/backend/access/aocs/aocsam.c

```diff
@@ -1268,7 +1298,11 @@ aocs_gettuple(AOCSScanDesc scan, int64 targrow, TupleTableSlot *slot)
 				/* continue next block */
 			}
 			else
-				pg_unreachable(); /* unreachable code */
+				ereport(ERROR,
+						(errcode(ERRCODE_INTERNAL_ERROR),
+						 errmsg("Unexpected EOF during analysis of AOCO table \'%s\' for column \'%s\'",
+								ds->ao_read.relationName,
+								scan->columnScanInfo.relationTupleDesc->attrs[attno].attname.data)));
```

Comment:
Can we be sure that `scan->columnScanInfo.relationTupleDesc` is not NULL here in all possible cases?

Reply:
Yes, this function is only called [here](https://github.com/arenadata/gpdb/blob/7355e32a20277c963e99141e6d84281fb04d3166/src/backend/access/aocs/aocsam.c#L1358), and [this](https://github.com/arenadata/gpdb/blob/7355e32a20277c963e99141e6d84281fb04d3166/src/backend/access/aocs/aocsam.c#L1335) ensures it is not NULL.